.github/workflows/publish.yml

```diff
@@ -176,48 +110,15 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          token: ${{ secrets.GH_TOKEN }}
           ref: ${{ inputs.branch }}
-      - uses: pnpm/action-setup@v4
-        with:
-          run_install: false
-      - uses: actions/setup-node@v5
-        with:
-          node-version-file: '.node-version'
-          cache: 'pnpm'
-
-      - name: Update npm
-        run: npm install -g npm@latest
-
-      - run: pnpm install --frozen-lockfile
 
-      - run: pnpm dlx nx-cloud start-ci-run --distribute-on=".nx/workflows/dynamic-changesets.yml" --stop-agents-after="e2e-ci" --with-env-vars="CODECOV_TOKEN"
-
-      - name: Cache Playwright browsers
-        uses: actions/cache@v4
-        with:
-          path: ~/.cache/ms-playwright
-          key: ${{ runner.os }}-playwright-${{ hashFiles('**/pnpm-lock.yaml') }}
-          restore-keys: |
-            ${{ runner.os }}-playwright-
-
-      - run: pnpm exec playwright install
-
-      - uses: nrwl/nx-set-shas@v4
-        with:
-          main-branch-name: main
-
-      - name: setup pnpm config
-        run: pnpm config set store-dir $PNPM_CACHE_FOLDER
-
-      - run: pnpm exec nx run-many -t build test e2e-ci
-
-      - uses: actions/upload-artifact@v4
-        if: ${{ !cancelled() }}
+      - name: Setup Project
+        uses: ./.github/actions/setup
         with:
-          name: playwright-report
-          path: |
-            ./**/.playwright/**
-          retention-days: 30
+          node-version-file: '.node-version'
+          pnpm-cache-folder: .pnpm-store
```

Comment:
nitpick but we can get rid of these inputs here and in the sub action. They're same as default.